download-lambda-logs/download_logs/aws_lambda.py

```diff
@@ -0,0 +1,76 @@
+import gzip, os, csv, requests, urllib.parse
+from io import TextIOWrapper
+
+import boto3
+s3 = boto3.client('s3')
+
+from .base import Base
+
+class AWSLambda(Base):
+    def __init__(self, bucket, filename):
+        self.bucket = bucket
+        super().__init__(filename)
+
+    def open_for_read(self):
+        print(self.bucket, self.filename)
+        obj = s3.get_object(Bucket=self.bucket, Key=self.filename)
+        return TextIOWrapper(gzip.GzipFile(fileobj=obj['Body'], mode='r'))
+
+    def process(self):
+        path = os.path.dirname(self.filename)
+        name = os.path.basename(self.filename)
+        date, rest = name.split('T', maxsplit=1)
+        y, m, d = date.split('-')
+
+        rest = rest.replace('.log.gz', '.json.gz')
+
+        csvreader = csv.reader(self.open_for_read(), delimiter="\t")
+
+        data = []
+        for row in csvreader:
+            result = self.transform_row(row)
+            if result != '' and result != None:
+                data.append(result)
+
+        return data  
+
+    def parse_row(self, row):
+        if row['ga_client_id'] != '':
+            row['ga_client_id'] = self.parse_client_id(row['ga_client_id'])
+        return row
+
+    def parse_client_id(self, client_id):
+        client_id = client_id.replace('GA', '')
+        client_id = client_id.split(".")[-2:]
+        return ".".join(client_id)
+
+    def send_event_to_GA(self, download_data):
+        PROPERTY_ID = 'UA-26179049-7'
+
+        params = urllib.parse.urlencode({
+                                        'v': 1,
+                                        'tid': PROPERTY_ID,
```

Comment:
As this is just a local variable, it should be downcased. Spelling this like a constant makes me want to look elsewhere in the file or in config for it